esp-hal/src/gpio/mod.rs

```diff
@@ -448,7 +448,7 @@ pub trait PeripheralOutput: PeripheralSignal {
 }
 
 /// Trait implemented by pins which can be used as inputs.
-pub trait InputPin: Pin + PeripheralInput {
+pub trait InputPin: Pin + PeripheralInput + Into<AnyPin> + 'static {
```

Comment:
These new bounds should have been added to `Pin` but that makes `RtcPin` no longer dyn compatible, which is a breaking change I do not wish to tackle here.